doc: matter: Update docs for the NCS 2.7.0 release (#16117)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/index.rst

   These changes are listed in the Matter fork section of the |NCS| :ref:`release_notes`.

.. important::

>    For Matter over Thread samples, starting with |NCS| releases after v2.6.0, the default cryptography backend is Arm PSA Crypto API instead of Mbed TLS, which was used in earlier versions.

Reviewer: ```suggestion
   For Matter over Thread samples, starting from |NCS| v2.6.0, the default cryptography backend is Arm PSA Crypto API instead of Mbed TLS, which was used in earlier versions.
```
after v2.6.0 isn't right.

Reviewer 2: Mbed TLS is used as a vague construct in this case, and in fact the most common source for PSA crypto APIs implemented (in software) is through the Mbed TLS deliverables.

I recommend we change the Mbed TLS to `m̀bedtls_`-prefixed crypto toolbox APIs from Mbed TLS

>    To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.

Reviewer 2: "backend" is a term we have moved away from as it was used for the `_ALT` based "driver support" that legacy crypto-toobox APIs. Can we fix this to just say "must use PSA Crypto APIs"?

Author: Sure, no problem, I will change it as you suggested. :)
